server/src/tasks.ts

```diff
@@ -31,40 +31,43 @@ export class TasksAndModels {
   }
 
   // Returns already saved model in priority, then the model from the task definition
-  private async loadModelFromTask (task: Task | TaskProvider): Promise<Model> {
+  private async loadModelFromTask(task: Task | TaskProvider): Promise<Model> {
     const discoTask = isTask(task) ? task : task.getTask()
     let model: Model | undefined
-
+    
     const modelPath = `./models/${discoTask.id}/`
     try {
       const content = await fs.readFile(`${modelPath}/model.json`)
+      console.log("Loading local model for task", discoTask.id)
       return await serialization.model.decode(content)
     } catch {
+      throw new Error("No local model found")
       // unable to read file, continuing
     }
-
-    if (isTask(task)) {
-      throw new Error('saved model not found and no way to get it')
-    } else {
-      model = await task.getModel()
-    }
-
-    await fs.mkdir(modelPath, { recursive: true })
-    const encoded = await serialization.model.encode(model)
-    await fs.writeFile(`${modelPath}/model.json`, encoded)
-
-    // Check digest if provided
-    if (discoTask.digest !== undefined) {
-      try {
-        await this.checkDigest(discoTask.digest, modelPath)
-      } catch (e) {
-        console.warn('removing nodel files at', modelPath)
-        await fs.rm(modelPath, { recursive: true, force: true })
-        throw e
-      }
-    }
-
-    return model
+    // console.log("Instantiating a new model", discoTask.id)
+    
+    // if (isTask(task)) {
+    //   throw new Error('saved model not found and no way to get it')
+    // } else {
+    //   model = await task.getModel()
+    // }
+    // console.log('Saving model to disk')
+    // await fs.mkdir(modelPath, { recursive: true })
+    // const encoded = await serialization.model.encode(model)
+    // await fs.writeFile(`${modelPath}/model.json`, encoded)
+    
+    // // Check digest if provided
+    // if (discoTask.digest !== undefined) {
+    //   try {
+    //     await this.checkDigest(discoTask.digest, modelPath)
+    //   } catch (e) {
+    //     console.warn('removing model files at', modelPath)
+    //     await fs.rm(modelPath, { recursive: true, force: true })
+    //     throw e
+    //   }
+    // }
+    
+    // return model
```

Comment:
avoid commented code

```suggestion
```

Reply:
Thanks! I think you reviewed an older commit? The code should be uncommented in the latest

Reply:
ho right, that was an old commit; nvm then